src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationContext.cs

```diff
@@ -82,17 +95,22 @@ internal sealed class CompilationContext
                     _locals,
                     inScopeHoistedLocalSlots,
                     _sourceMethodParametersInOrder,
-                    out var displayClassVariableNamesInOrder,
+                    parametersAreOutside: currentFrame.ParameterCount == 0,
```

Comment:
>currentFrame.ParameterCount == 0 [](http://example.com/codeflow?start=42&length=32)

Why does `ParameterCount == 0` determine whether parameters are considered outside of inside?

Reply:
> This is a special situation for async MoveNext method.

Consider adding a comment here to indicate that.